pkg/plugins/golang/v2/api.go

```diff
@@ -126,29 +128,29 @@ func (p *createAPISubcommand) Run() error {
 }
 
 func (p *createAPISubcommand) Validate() error {
-	if err := p.resource.ValidateV2(); err != nil {
+	if err := p.options.Validate(); err != nil {
 		return err
 	}
 
 	reader := bufio.NewReader(os.Stdin)
 	if !p.resourceFlag.Changed {
 		fmt.Println("Create Resource [y/n]")
-		p.doResource = util.YesNo(reader)
+		p.options.DoAPI = util.YesNo(reader)
 	}
 	if !p.controllerFlag.Changed {
 		fmt.Println("Create Controller [y/n]")
-		p.doController = util.YesNo(reader)
+		p.options.DoController = util.YesNo(reader)
 	}
 
 	// In case we want to scaffold a resource API we need to do some checks
-	if p.doResource {
+	if p.options.DoAPI {
```

Comment:
Why we need the options at all? Why not we can have  p.resource as before? 
See that you will get all options to create the resource at the same way.